Go modules

tzdybal opened the pull request:
This PR replaces `env.sh` script with Go Modules. It's mostly backward compatible:
 - you can build in `$GOPATH/src/github.com/ethereum/go-ethereum`
 - you can build in any other directory outside `$GOPATH`

The one "incompatibility" is:
 - to build in `$GOPATH/src/github.com/multi-geth/multi-geth` you need to set environment variable `GO111MODULES=on` - this will be default behavior in Go 1.13.

This PR resolves #41.

Hopefully this changes will be accepted as is in upstream repository.

tzdybal committed:
go.mod: Go modules initialization

tzdybal committed:
Makefile,go.*,build: use Go modules

Approach with `_workspace` directory was replaced with Go modules.
gometalinter was updated to v3.0.0.

tzdybal committed:
build: enable modules before getting gometalinter

To keep things simple - force using modules to install specific version
of gometalinter. Without this, current working directory and GO111MODULE
environment variable must be used to decide whether to use versioned or
unversioned `go get`.

meowsbits commented on build/ci.go:
```diff
@@ -357,8 +357,10 @@ func doLint(cmdline []string) {
-	build.MustRun(goTool("get", "gopkg.in/alecthomas/gometalinter.v2"))
-	build.MustRunCommand(filepath.Join(GOBIN, "gometalinter.v2"), "--install")
+	os.Setenv("GO111MODULE", "on")
+	build.MustRun(goTool("get", "gopkg.in/alecthomas/kingpin.v3-unstable@63abe20a23e29e80bbef8089bd3dee3ac25e5306"))
```
What's the reason for this version bump?

tzdybal replied:
I like to use up to date software.

meowsbits replied:
lol. weird.

tzdybal committed:
build/ci.go: different options for gometalinter

tzdybal committed:
build/ci.go: execute gometalinter without modules

tzdybal committed:
vendor: remove vendored packages

tzdybal committed:
crypto,les,p2p,tests: flags workaround in tests

This change is releated to: https://github.com/golang/go/issues/31859

tzdybal committed:
accounts: updated pcsc API call

tzdybal committed:
build: switch to golangci-lint

tzdybal committed:
go.*: updated go.mod and go.sum files

tzdybal committed:
.travis.yml: update to go v1.13.x

tzdybal committed:
internal/build: ensure modules are downloaded before expanding packages

tzdybal committed:
accounts: use local directory for bind tests

Because of go modules, it's easiest to create tests in local
subdirectory, to ensure that correct version of dependencies is used.

tzdybal committed:
all: fix linter errors

tzdybal committed:
go.mod,go.sum: updated go modules files
